rotkehlchen/chain/manager.py

```diff
@@ -824,384 +829,49 @@ def sync_bitcoin_accounts_with_db(
             if account not in db_btc_accounts:
                 accounts_to_remove.append(account)
 
-        if blockchain == SupportedBlockchain.BITCOIN:
-            balances_mapping = get_bitcoin_addresses_balances(accounts_to_remove)
-        else:
-            balances_mapping = get_bitcoin_cash_addresses_balances(accounts_to_remove)
-        balances = [balances_mapping.get(x, ZERO) for x in accounts_to_remove]
         self.modify_blockchain_accounts(
             blockchain=blockchain,
             accounts=accounts_to_remove,
             append_or_remove='remove',
-            add_or_sub=operator.sub,
-            already_queried_balances=balances,
-        )
-
-    def modify_btc_account(
-            self,
-            account: BTCAddress,
-            append_or_remove: str,
-            add_or_sub: AddOrSub,
-            already_queried_balance: Optional[FVal] = None,
-    ) -> None:
-        """Either appends or removes a BTC acccount.
-
-        If already_queried_balance is not None then instead of querying the balance
-        of the account we can use the already queried one.
-
-        Call with 'append', operator.add to add the account
-        Call with 'remove', operator.sub to remove the account
-
-        May raise:
-        - RemotError if there is a problem querying blockchain.info or cryptocompare
-        """
-        btc_usd_price = Inquirer().find_usd_price(A_BTC)
-        remove_with_populated_balance = (
-            append_or_remove == 'remove' and len(self.balances.btc) != 0
-        )
-        # Query the balance of the account except for the case when it's removed
-        # and there is no other account in the balances
-        if append_or_remove == 'append' or remove_with_populated_balance:
-            if already_queried_balance is None:
-                balances = get_bitcoin_addresses_balances([account])
-                balance = balances[account]
-            else:
-                balance = already_queried_balance
-            usd_balance = balance * btc_usd_price
-
-        if append_or_remove == 'append':
-            self.balances.btc[account] = Balance(amount=balance, usd_value=usd_balance)
-        elif append_or_remove == 'remove':
-            if account in self.balances.btc:
-                del self.balances.btc[account]
-        else:
-            raise AssertionError('Programmer error: Should be append or remove')
-
-        if len(self.balances.btc) == 0:
-            # If the last account was removed balance should be 0
-            self.totals.assets[A_BTC] = Balance()
-        else:
-            self.totals.assets[A_BTC] = add_or_sub(
-                self.totals.assets[A_BTC],
-                Balance(balance, usd_balance),
-            )
```

Comment:
You have also deleted this from all the chain deletions. This still needs to be there. You need to adjust the totals at each deletion.

And ... as I am writing this I realize this is hard without a query at removal. So for now you probably need to query the balance at deletion just to be able to subtract from the totals.

-----

But need to think a bit about this way of keeping balances. The totals should be calculatable at any point from all the rest so we should not have to do this. But this is a different refactor.

Reply:
> But need to think a bit about this way of keeping balances. The totals should be calculatable at any point from all the rest so we should not have to do this. But this is a different refactor.

So spent some time looking at this. And can't find a reason why we do the totals in the chains like this.

I may still be missing something but perhaps we can simply not query balances at removal and instead have a `recalculate_totals` method of `BlockchainBalances` which should be called each time (1) balances are queried and (2) address is removed.

Reply:
Okay, will do it this way